### PR TITLE
TaxonomyManager: Hide View Posts link for jetpack sites.

### PR DESCRIPTION
**Pull request by timmyc**
For #9927 - The "View Posts" link for a term in the Taxonomy Manager assumes the taxonomy slug ( i.e. `category` ) can be used to display a post listing by term on the website.  In Jetpack/.org sites though, a site can customize a `category_base` and `tag_base`.  

Currently, we are not returning the `category_base` or `tag_base` from the site endpoints so we unfortunately do not have a way to properly link to these Jetpack sites that have customized these "term base" options.

I _believe_ this might be the reason why the "Tags & Categories" module on the Stats Insights page is hidden on Jetpack sites too.

Instead of letting this issue hold up the release of the Taxonomy Manager, I'm opting to hide this "View Posts" link on Jetpack sites within Calypso until we have access to the appropriate site options and can correctly render these links.

__To Test__
- Open either categories or tags from the Site Settings > Writing page for a WPCOM site
- Click the ellipsis menu for a term, verify the View Posts link is still shown
- Do the same on a Jetpack site, but verify the View Posts link is not shown

### CODE REVIEW

File: client/blocks/taxonomy-manager/list-item.jsx [hunk 1]
```diff
@@ -156,9 +157,11 @@ class TaxonomyManagerListItem extends Component {
 							{ translate( 'Delete' ) }
 						</PopoverMenuItem>
 					}
-					<PopoverMenuItem href={ this.getTaxonomyLink() } icon="external">
-						{ translate( 'View Posts' ) }
-					</PopoverMenuItem>
+					{ ! isJetpack &&
+						<PopoverMenuItem href={ this.getTaxonomyLink() } icon="external">
```
Comment (youknowriad) [inline client/blocks/taxonomy-manager/list-item.jsx]: Well! this is not directly related but I think we could move the logic contained in `getTaxonomyLink` to `state/sites/selectors` and we can test it properly. This function could return a falsy value for Jetpack sites. The component here won't have any specific JetPack logic but just checks the falsy value instead. Thoughts on this approach ?

Comment (timmyc) [inline client/blocks/taxonomy-manager/list-item.jsx]: I like this idea, but should it be a term selector since it is a term oriented link?  I'll just do a follow-up PR though.  I did forget to mention that I created an API diff to add the needed values to the site response in code-D3665
